### DISCUSSION TITLE
What's the plan for `Dynamic` in TurboModules?

### DISCUSSION THREAD

**Discussion by bnickel** [reactwg/react-native-new-architecture#208] [Q&A] [answered]
A customer of ours reported that their Android app crashed on start with the message in logcat:

```
Error: Exception in HostObject::get for prop 'HeapReactNativeBridge':
com.facebook.react.internal.turbomodule.core.TurboModuleInteropUtils$ParsingException:
Unable to parse @ReactMethod annotation from native module method: HeapReactNativeBridge.handleResult().
Details: Unable to parse JNI signature. Detected unsupported parameter class: com.facebook.react.bridge.Dynamic`
```

Upon investigation, I confirmed that this is coming from new architecture because Dynamic isn't currently supported for parameters.

```
    if (paramClass == Dynamic.class) {
      // TODO(T145105887): Output warnings that TurboModules doesn't yet support Dynamic arguments
    }
```

https://github.com/facebook/react-native/blob/c8a1da4c9e8bda9569098baf6f1daf846df77383/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/internal/turbomodule/core/TurboModuleInteropUtils.java#L168-L170

Is there an expected timeframe for either (A) supporting `Dynamic` parameters or (B) more gracefully handling a module with an unsupported type?

**Comment by cortinico** [comment, marked as answer]
> Is there an expected timeframe for either (A) supporting `Dynamic` parameters or (B) more gracefully handling a module with an unsupported type?

I'll be able to give you a more precise answer next week about (A). I'll look into it and get back to you if we'll be able to support it both for Turbo Modules and for the Interop Layer.

We have a partner which is blocked by this for a migration, so this is quite important for us.

If (A) is not an option, we can definitely implement (B). Still the app will have to crash if you attempt to invoke a method with an unsupported parameter.

**Reply by cortinico** [reply]
Small update: we do have a PR out with support for `Dynamic`
- https://github.com/facebook/react-native/pull/45944

I'll keep you posted on when/how we plan to merge it
